sky/clouds/service_catalog/common.py

```diff
@@ -489,7 +489,13 @@ def get_accelerators_from_instance_type_impl(
     acc_name, acc_count = row['AcceleratorName'], row['AcceleratorCount']
     if pd.isnull(acc_name):
         return None
-    return {acc_name: int(acc_count)}
+
+    def _convert(value):
+        if int(value) == value:
+            return int(value)
+        return float(value)
+
+    return {acc_name: _convert(acc_count)}
 
 
 def get_instance_type_for_accelerator_impl(
```

Comment:
Should we update the `acc_count` type here? Also, when we are comparing the `acc_count` should we make sure every number within `math.abs(df['AcceleratorCount'] - acc_count) <= 0.01` should work. Otherwise, a user running `sky launch --gpus A10:0.16` or `sky launch --gpus A10:0.1666` would fail?

Reply:
> Should we update the acc_count type here?

Sry could you elaborate on this...? Are you saying there are a better place to update the type or what..?

> Also, when we are comparing the `acc_count` should we make sure every number within `math.abs(df['AcceleratorCount'] - acc_count) <= 0.01` should work.

For this, I'm slightly concerned about the case when user:

```bash
sky launch -c a10-frac --gpus A10:0.16 # detected as 0.167 in catalog, then launch cluster with 0.167 gpu
sky exec a10-frac --gpus A10:0.16 sleep 100000 # user would think the cluster is full
sky exec a10-frac --gpus A10:0.007 sleep 100000 # however, this still running as the cluster is actually launched with 0.167 gpu.
```

To deal with the failing, we currently shows all valid instance type as fuzzy candidate and user could then modify their acc count:

```bash
sky launch --gpus A10:0.16
I 09-12 22:34:50 optimizer.py:1301] No resource satisfying <Cloud>({'A10': 0.16}) on [AWS, GCP, Azure, RunPod].
I 09-12 22:34:50 optimizer.py:1305] Did you mean: ['A100-80GB-SXM:1', 'A100-80GB-SXM:2', 'A100-80GB-SXM:4', 'A100-80GB-SXM:8', 'A100-80GB:1', 'A100-80GB:2', 'A100-80GB:4', 'A100-80GB:8', 'A100:1', 'A100:16', 'A100:2', 'A100:4', 'A100:8', 'A10:0.167', 'A10:0.333', 'A10:0.5', 'A10:1', 'A10:2', 'A10G:1', 'A10G:4', 'A10G:8']
sky.exceptions.ResourcesUnavailableError: Catalog does not contain any instances satisfying the request:
Task<name=sky-cmd>(run=<empty>)
  resources: <Cloud>({'A10': 0.16}).

To fix: relax or change the resource requirements.
Try one of these offered accelerators: ['A100-80GB-SXM:1', 'A100-80GB-SXM:2', 'A100-80GB-SXM:4', 'A100-80GB-SXM:8', 'A100-80GB:1', 'A100-80GB:2', 'A100-80GB:4', 'A100-80GB:8', 'A100:1', 'A100:16', 'A100:2', 'A100:4', 'A100:8', 'A10:0.167', 'A10:0.333', 'A10:0.5', 'A10:1', 'A10:2', 'A10G:1', 'A10G:4', 'A10G:8']

Hint: sky show-gpus to list available accelerators.
      sky check to check the enabled clouds.
```

Does that sounds good to you?

Reply:
> acc_count is currently int for the type annotation below. Should we update that?
Done!
> This will only apply for the case when a user is actually creating an instance with A10:0.16 for this function right? When we are launching an instance, once we returned the instance type, we can round up the request to the actual acc_count in the catalog?
Good point! Done. PTAL!